### PR TITLE
IEP-446: IDF Components Install Page and Manager (Updated Branch & PR)

### PR DESCRIPTION
**Pull request by alirana01**

### CODE REVIEW

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/installcomponents/handler/InstallCommandHandler.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+		}
+		Map<String, String> envMap = new IDFEnvironmentVariables().getEnvMap();
+		IProject selectedProject = EclipseUtil.getSelectedProjectInExplorer();
+
+		Path pathToProject = new Path(selectedProject.getLocation().toString());
+		List<String> commands = new ArrayList<>();
+		commands.add(IDFUtil.getIDFPythonEnvPath());
+		commands.add(IDFUtil.getIDFPythonScriptFile().getAbsolutePath());
+		commands.add("add-dependency"); //$NON-NLS-1$
+		commands.add(namespace.concat("/").concat(name.concat("==").concat(version))); //$NON-NLS-1$ //$NON-NLS-2$
```
Comment (sigmaaa) [inline bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/installcomponents/handler/InstallCommandHandler.java:58]: what if the component's version is not specified?

**Review by sigmaaa — CHANGES_REQUESTED** [review]
Looks like

> add-dependency

command was added in the ESP-IDF 4.4 (I can't use install button on 4.3). We can mention this in documentation. 
Also, I'm getting an error, when i'm trying to install components with more then one symbol "_"or "-" at the component's name. For example on 

> diamond_dependecy_c:

```
Executing action: add-dependency
Invalid dependency version requirement: _c>=1. Please use format like ">=1" or "*". 
```

**Comment by kolipakakondal** [comment]
Hi @alirana01  @sigmaaa  Can we merge this?

**Comment by alirana01** [comment]
@kolipakakondal merged into master

**Pull request merged**